src/register/RegistrationPage.jsx

```diff
@@ -177,8 +178,16 @@ const RegistrationPage = (props) => {
 
       // This is used by the "User Retention Rate Event" on GTM
       setCookie(getConfig().USER_RETENTION_COOKIE_NAME, true);
+
+      const { country } = configurableFormFields;
+      if (country?.countryCode !== IPBasedCountryCode) {
+        sendTrackEvent('edx.bi.user.auto-populatedcountry.changed', {
+          originalCountry: IPBasedCountryCode,
+          selectedCountry: country.countryCode,
+        });
+      }
     }
-  }, [registrationResult]);
+  }, [registrationResult, IPBasedCountryCode, configurableFormFields]);
```

Comment:
Just pick country in the dependency array. 
```suggestion
  }, [registrationResult, IPBasedCountryCode, configurableFormFields.country]);
```

Reply:
addressed

Reply:
I have tried to use it but there is lint error which says ```configurableFormFields``` is not in a dependency array

Reply:
Remove line number 185 and directly us the const `configurableFormFields.country` value

Reply:
yeah that makes sense